azbus/receiver.go

```diff
@@ -82,22 +105,39 @@ type Receiver struct {
 
 	Cfg ReceiverConfig
 
-	log      Logger
-	mtx      sync.Mutex
-	receiver *azservicebus.Receiver
-	options  *azservicebus.ReceiverOptions
-	handlers []Handler
+	log          Logger
+	mtx          sync.Mutex
+	receiver     *azservicebus.Receiver
+	options      *azservicebus.ReceiverOptions
+	handlers     []Handler
+	batchHandler BatchHandler
```

Comment:
I think we should have separate type for `BatchReceiver` tbh, as what does it even mean to mix the two here?

Reply:
I'll have a crack at that. I agree it would be better